src/libs/metadata_view.c

```diff
@@ -791,6 +807,115 @@ void connect_key_accels(dt_lib_module_t *self)
   dt_accel_connect_lib(self, "jump to film roll", closure);
 }
 
+static void _update_layout(dt_lib_module_t *self)
+{
+  dt_lib_metadata_view_t *d = (dt_lib_metadata_view_t *)self->data;
+  char *pref = dt_conf_get_string("plugins/lighttable/metadata_view/pref_hidden");
+  GList *prefs = dt_util_str_to_glist(",", pref);
+  g_free(pref);
+  for(int i = 0; i < md_size; i++)
+  {
+    const gboolean hidden = g_list_find_custom(prefs, _labels[i], (GCompareFunc)g_strcmp0) != NULL;
+    for(int j = 0; j < 2; j++)
+    {
+      GtkWidget *w = gtk_grid_get_child_at(GTK_GRID(d->grid), j, i);
+      gtk_widget_show_all(w);
+      gtk_widget_set_no_show_all(w, TRUE);
+      gtk_widget_set_visible(w, !hidden);
+    }
+  }
+  g_list_free_full(prefs, g_free);
+}
+
+void set_preferences(dt_lib_module_t *self)
+{
+  dt_lib_metadata_view_t *d = (dt_lib_metadata_view_t *)self->data;
+
+  GtkWidget *win = dt_ui_main_window(darktable.gui->ui);
+  GtkWidget *dialog = gtk_dialog_new_with_buttons(_("metadata settings"), GTK_WINDOW(win), GTK_DIALOG_DESTROY_WITH_PARENT,
+                                       _("cancel"), GTK_RESPONSE_NONE, _("save"), GTK_RESPONSE_YES, NULL);
+  GtkWidget *area = gtk_dialog_get_content_area(GTK_DIALOG(dialog));
+
+  GtkWidget *w = gtk_scrolled_window_new(NULL, NULL);
+  gtk_widget_set_size_request(w, -1, DT_PIXEL_APPLY_DPI(300));
+  gtk_scrolled_window_set_policy(GTK_SCROLLED_WINDOW(w), GTK_POLICY_AUTOMATIC, GTK_POLICY_AUTOMATIC);
+  gtk_container_add(GTK_CONTAINER(area), w);
+
+  GtkWidget *grid = gtk_grid_new();
+  gtk_container_add(GTK_CONTAINER(w), grid);
+  GtkWidget *label = gtk_label_new(_("metadata"));
+  gtk_grid_attach(GTK_GRID(grid), label, 0, 0, 1, 1);
+  label = gtk_label_new(_("hidden"));
+  gtk_grid_attach(GTK_GRID(grid), label, 1, 0, 1, 1);
+  gtk_widget_set_tooltip_text(GTK_WIDGET(label),
+            _("tick if the corresponding metadata is of no interest for you"));
+
+  char *pref = dt_conf_get_string("plugins/lighttable/metadata_view/pref_hidden");
+  GList *prefs = dt_util_str_to_glist(",", pref);
+  g_free(pref);
+  GtkWidget *hidden[md_size];
+  gboolean hide[md_size];
+  for(int i = 0; i < md_size; i++)
+  {
+    label = gtk_label_new(_md_labels[i]);
+    gtk_widget_set_halign(label, GTK_ALIGN_START);
+    gtk_label_set_xalign (GTK_LABEL(label), 0.0f);
+    gtk_label_set_ellipsize(GTK_LABEL(label), PANGO_ELLIPSIZE_END);
+    gtk_grid_attach(GTK_GRID(grid), label, 0, i+1, 1, 1);
+
+    hidden[i] = gtk_check_button_new();
+    gtk_widget_set_halign(hidden[i], GTK_ALIGN_CENTER);
+    hide[i] = g_list_find_custom(prefs, _labels[i], (GCompareFunc)g_strcmp0) != NULL;
+    gtk_toggle_button_set_active(GTK_TOGGLE_BUTTON(hidden[i]), hide[i]);
+    gtk_grid_attach(GTK_GRID(grid), hidden[i], 1, i+1, 1, 1);
+    if(i >= md_xmp_metadata && i < md_xmp_metadata + DT_METADATA_NUMBER)
```

Comment:
If you moved `gtk_widget_show_all(dialog);` _before_ the loop, you could simplify this to
```
    // metadata plugin preferences have precedence
    gboolean hide_line = i >= md_xmp_metadata &&
                         i < md_xmp_metadata + DT_METADATA_NUMBER &&
                         !gtk_widget_is_visible(GTK_WIDGET(d->name[i]));
    gtk_widget_set_visible(label, !hide_line);
    gtk_widget_set_visible(hidden[i], !hide_line);

```
but I'm not sure about the logic and it also doesn't seem to work:
- even if I myself am not normally using particular metadata fields (so I don't want them in the editor) I might still want to see what somebody else put in there or if the field contains rubbish. However at the moment I cannot select them in the info module if I don't have it in the metadata module.
- when a metadata field _is_ visible in the metadata module, I can chose to hide it in the info module (which makes sense; I can already see it on the right side after all). However it will reappear as soon as I mouseover a different image.

Reply:
>     * I might still want to see what somebody else put in there or if the field contains rubbish. However at the moment I cannot select them in the info module if I don't have it in the metadata module.

I've made that on purpose, but I understand your point. I'll remove the precedence.

>     * when a metadata field _is_ visible in the metadata module, I can chose to hide it in the info module (which makes sense; I can already see it on the right side after all). However it will reappear as soon as I mouseover a different image.

Good catch. I'll fix that.

> If you moved `gtk_widget_show_all(dialog);`

I've been lasy here and copied the piece of code you gave me for metadata. I'd already tried to to put that on the grid but it did not work. Am I wrong on this ? Is there any reason ? In such case I'll do it also in metadata.

Reply:
Precedence removed and reported issue solved.